batch2/create-batch-tables.sql

```diff
@@ -274,6 +351,7 @@ BEGIN
     UPDATE jobs SET state = 'Ready', instance_name = NULL WHERE batch_id = in_batch_id AND job_id = in_job_id;
     UPDATE ready_cores SET ready_cores_mcpu = ready_cores_mcpu + cur_cores_mcpu;
     UPDATE instances SET free_cores_mcpu = free_cores_mcpu + cur_cores_mcpu WHERE name = cur_job_instance_name;
+    UPDATE attempts SET end_time = new_end_time, reason = new_reason WHERE batch_id = in_batch_id AND job_id = in_job_id;
```

Comment:
Whoa, this can't be right.  You're setting the end time for all the attempts, some of which might have been preempted a long time ago.

Reply:
I was assuming the same logic where the minimum end time is always taken. I can do a join with the jobs table, filter to the attempts on that instance and update or go further and look for attempts where the reason is not completed, preempted.

Reply:
@cseed I'm not sure what to do here. The more complicated query doesn't buy you anything because the job is already checked to be on that instance_id with the if clause. I think my query is fine.

Reply:
I think we should store the attempt in the job.  Then you can get the cur_attempt when you query the job, and then reset just that one.